component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,409 @@
+        public void onFileDelete(final File file) {
+            log.debug(file.getAbsolutePath() + " was deleted.");
+            sourceEventListener.onEvent
+                    (getFileHandlerEvent(file, listeningFileUri, STATUS_DONE), null);
+        }
+
+        @Override
+        public void onStop(final FileAlterationObserver observer) {
+        }
+    }
+
+    public void initiateFileAlterationObserver() {
+        FileAlterationObserver observer = new FileAlterationObserver(listeningDirUri);
+        observer.addListener(new FileAlterationImpl());
+        monitor = new FileAlterationMonitor(monitoringInterval);
+        monitor.addObserver(observer);
+        try {
+            monitor.start();
+            log.info("Directory monitoring has been started for folder : " + uri + " .");
+        } catch (Exception e) {
+            throw new SiddhiAppRuntimeException("Exception occurred when starting server to monitor " + uri + ".", e);
+        }
+    }
+
+    @Override
+    public void disconnect() {
+
+        if (monitor != null) {
+            try {
+                monitor.stop();
+                initialMap.clear();
+                log.info("Directory monitoring has been stopped for folder : " + uri + " .");
+            } catch (Exception e) {
+                log.error("Exception occurred when stopping server  while monitoring " + uri + " .", e);
+            }
+        }
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    @Override
+    public void pause() {
+        if (monitor != null) {
+            threadList.forEach(SubThread::pause);
+            log.info("Directory monitoring has been paused for folder : " + uri + " .");
+        }
+    }
+
+    @Override
+    public void resume() {
+        if (monitor != null) {
+            threadList.forEach(SubThread::resume);
+            log.info("Directory monitoring has been resumed for folder : " + uri + " .");
+        }
+    }
+
+    /**
+     * Implementation of SubThread
+     */
+    public class SubThread implements Runnable {
+        String status;
+        private final File file;
+        private volatile boolean paused;
+        private ReentrantLock lock;
+        private Condition condition;
+        private volatile boolean inactive;
+
+        SubThread(File file) {
+            this.file = file;
+            inactive = false;
+            lock = new ReentrantLock();
+            condition = lock.newCondition();
+        }
+
+        @Override
+        public void run() {
+            while (!inactive) {
+                if (paused) {
+                    lock.lock();
+                    try {
+                        condition.await();
+                    } catch (InterruptedException ie) {
+                        Thread.currentThread().interrupt();
+                    } finally {
+                        lock.unlock();
+                    }
+                }
+                long val = initialMap.get(file.getAbsolutePath());
```

Comment:
lets rename 'val' to 'recentFileSize' or a similar.

Reply:
fixed in 3f9a2a0